components/camel-seda/src/main/java/org/apache/camel/component/seda/SedaEndpoint.java

```diff
@@ -213,8 +215,14 @@ protected BlockingQueue<Exchange> createQueue() {
      * @return the reference, or <tt>null</tt> if no queue reference exists.
      */
     public synchronized QueueReference getQueueReference() {
-        String key = getComponent().getQueueKey(getEndpointUri());
-        QueueReference ref = getComponent().getQueueReference(key);
+        if (key == null) {
```

Comment:
I wonder if this can be done in doStart - I assume getEndpointUri is static at the moment when its started, and maybe then this can be computed once, and then the synchronization may not be need at all

Reply:
🤔 Thanks! Sounds like a good idea and I can give it a try. 

I'll dispatch the other PRs I have on the queue and then I'll go back to this one.

Reply:
It turns out we can do that. I updated the PR with your suggestion, thanks!